packages/shared/src/components/modals/NewSquadModal.tsx

```diff
@@ -44,29 +47,32 @@ function NewSquadModal({
     onRequestClose,
   };
 
-  const modalSteps: ModalStep[] = [
-    {
-      key: ModalState.Details,
-      title: (
-        <>
-          {onPreviousState && (
-            <Modal.Header.StepsButton onClick={onPreviousState} />
-          )}
-          <Modal.Header.Subtitle>{ModalState.Details}</Modal.Header.Subtitle>
-        </>
-      ),
-    },
-    {
-      key: ModalState.SelectArticle,
-    },
-    {
-      key: ModalState.WriteComment,
-    },
-    {
-      key: ModalState.Ready,
-      title: <Modal.Header.Title>{ModalState.Ready}</Modal.Header.Title>,
-    },
-  ];
+  const modalSteps = useMemo(() => {
+    const steps: ModalStep[] = [
```

Comment:
Here is a cleaner way to achieve this

```
  const modalSteps: ModalStep[] = [
    {
      key: ModalState.Details,
      title: (
        <>
          {onPreviousState && (
            <Modal.Header.StepsButton onClick={onPreviousState} />
          )}
          <Modal.Header.Subtitle>{ModalState.Details}</Modal.Header.Subtitle>
        </>
      ),
    },
    !post ? {
      key: ModalState.SelectArticle,
    } : undefined,
    {
      key: ModalState.WriteComment,
    },
    {
      key: ModalState.Ready,
      title: <Modal.Header.Title>{ModalState.Ready}</Modal.Header.Title>,
    },
  ].filter(step => step);
```

Reply:
Nice! Love this refactor! Let me update it quickly!